mag/mag-als31300.c

```diff
@@ -0,0 +1,413 @@
+/*
+ * Phoenix-RTOS
+ *
+ * ALS31300 i2c Magnetometer Sensor driver
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Marek Bialowas
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <unistd.h>
+#include <sys/minmax.h>
+
+#include <sys/msg.h>
+#include <posix/utils.h>
+
+#include <i2c.h>
+
+#define LOG_TAG       "mag-als31300: "
+#define log(fmt, ...) printf(LOG_TAG fmt "\n", ##__VA_ARGS__)
+
+
+#define I2C_DEV_ADDR 0x60u
+#define CFG_REG1     0x02u
+#define DATA_REG1    0x28u
+#define DATA_REG2    0x29u
+
+#define CFG_REG1_I2C_ADDR_MASK    0x0001FC00u
```

Comment:
nitpick - we use lowercase hex numbers

Reply:
daamn, we use uppercase numbers, confusing.